{
  try {
    handleMessage(message);
  }
 catch (  Exception e) {
  }
}
